enumerables.rb

```diff
@@ -1,5 +1,107 @@
-  module Enumerable
+module Enumerable
     def my_each
-      # your code here
+      self.length.times do |x|
+        yield(self[x])
+      end
     end
-  end
\ No newline at end of file
+
+
+    def my_each_with_index
+      index = 0
+      self.length.times do |x|
+        yield(index)
+        index += 1
+      end
+    end
+
+
+    def my_select
+      arr = Array.new
+      self.length.times do |x|
+        if yield(self[x])
+          arr.push(x)
+        end
+      end
+      arr
+    end
+    
+
+    def my_all?
+      self.length.times do |x|
+        if !yield(self[x])
+          return false
+        else 
+          return true
+        end
+      end
+    end
+
+
+    def my_any?
+      self.length.times do |x|
+        if yield(self[x]) != false
+          return true
+        end
+      end
+      false
+    end
+
+
+    def my_none?
+      self.length.times do |x|
+        if yield(self[x]) 
+          return false
+        end
+      end
+      true
+    end
+
+
+    def my_count
+      index = 0
+      self.length.times do |x|
+        # yield(index)
+        index += 1
+      end
+      index
+    end
+
+
+    def my_map(proc = nil)
+      arr = Array.new
+      if proc 
+        self.length.times do |x|
+          arr.push(proc.call(self[x]))
+        end
+      else 
+        self.length.times do |x|
+          arr.push(yield(self[x]))
+        end
+      end
+      arr
+    end
+
+
+    def my_inject
```

Comment:
this method does not return the proper return and it should also have a default parameter
It should be something like this
```ruby 

    def my_inject(n = 0)
      out = n
      self.my_each do |x|
        output = yield(out, x)
      end
      return out
    end
```